autogluon/utils/tabular/ml/models/fastainn/tabular_nn_fastai.py

```diff
@@ -0,0 +1,345 @@
+        from fastai.utils.mod_display import progress_disabled_ctx
+        from .callbacks import EarlyStoppingCallbackWithTimeLimit
+
+        start_time = time.time()
+
+        self.y_scaler = self.params.get('y_scaler', None)
+        if self.y_scaler is not None:
+            self.y_scaler = copy.copy(self.y_scaler)
+
+        logger.log(15, f'Fitting Neural Network with parameters {self.params}...')
+        data = self.preprocess_train(X_train, y_train, X_val, y_val)
+
+        nn_metric, objective_func_name = self.__get_objective_func_name()
+        objective_func_name_to_monitor = self.__get_objective_func_to_monitor(objective_func_name)
+        objective_optim_mode = 'min' if objective_func_name in [
+            'root_mean_squared_error', 'mean_squared_error', 'mean_absolute_error', 'r2'  # Regression objectives
+        ] else 'auto'
+
+        # TODO: calculate max emb concat layer size and use 1st layer as that value and 2nd in between number of classes and the value
+        if self.params.get('layers', None) is not None:
+            layers = self.params['layers']
+        elif self.problem_type in [REGRESSION, BINARY]:
+            layers = [200, 100]
+        else:
+            base_size = max(len(data.classes) * 2, 100)
+            layers = [base_size * 2, base_size]
+
+        loss_func = None
+        if self.problem_type in [BINARY, MULTICLASS] and self.params.get('smoothing', 0.0) > 0.0:
+            loss_func = LabelSmoothingCrossEntropy(self.params['smoothing'])
+
+        ps = self.params['ps']
+        if type(ps) != list:
+            ps = [ps]
+
+        if time_limit:
+            time_elapsed = time.time() - start_time
+            time_left = time_limit - time_elapsed
+        else:
+            time_left = None
+
+        early_stopping_fn = partial(EarlyStoppingCallbackWithTimeLimit, monitor=objective_func_name_to_monitor, mode=objective_optim_mode,
+                                    min_delta=self.params['early.stopping.min_delta'], patience=self.params['early.stopping.patience'], time_limit=time_left)
+
+        self.model = tabular_learner(
+            data, layers=layers, ps=ps, emb_drop=self.params['emb_drop'], metrics=nn_metric,
+            loss_func=loss_func, callback_fns=[early_stopping_fn]
+        )
+        logger.log(15, self.model.model)
+
+        with make_temp_directory() as temp_dir:
+            save_callback = SaveModelCallback(self.model, monitor=objective_func_name_to_monitor, mode=objective_optim_mode, name=self.name)
+            with progress_disabled_ctx(self.model) as model:
+                original_path = model.path
+                model.path = Path(temp_dir)
+                model.fit_one_cycle(self.params['epochs'], self.params['lr'], callbacks=save_callback)
+
+                # Load the best one and export it
+                model.load(self.name)
+
+                if objective_func_name == 'log_loss':
+                    self.eval_result = model.validate()[0]
```

Comment:
self.eval_result? What is this, why is it a variable not defined in init?

Reply:
removed `self.`: leftovers from old code